core/domain/question_domain.py

```diff
@@ -203,9 +203,39 @@ def _convert_state_v28_dict_to_v29_dict(cls, question_state_dict):
         question_state_dict['solicit_answer_details'] = False
         return question_state_dict
 
+    @classmethod
+    def _convert_state_v29_dict_to_v30_dict(
+            cls, question_state_dict, linked_skill_ids):
+        """Converts from version 29 to 30. Version 30 replaces
+        tagged_misconception_id with tagged_skill_misconception_id, which
+        contains the skill id and misconception id of the tagged misconception,
+        connected by '-'.
+
+        Args:
+            question_state_dict: dict. A dict where each key-value pair
+                represents respectively, a state name and a dict used to
+                initalize a State domain object.
+            linked_skill_ids: list(str). Skill ids linked to the question.
+
+        Returns:
+            dict. The converted question_state_dict.
+        """
+        answer_groups = question_state_dict['interaction']['answer_groups']
+        for answer_group in answer_groups:
+            if answer_group['tagged_misconception_id'] != None:
+                answer_group['tagged_skill_misconception_id'] = (
+                    linked_skill_ids[0] + '-' +
```

Comment:
If we only need the 1st linked skill then is there any specific reason of sending the list?

Reply:
Changed it to the first linked skill only.